core/src/main/java/nu/yona/server/subscriptions/service/MigratePrivateUserDataService.java

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Stichting Yona Foundation This Source Code Form is subject to the terms of the Mozilla Public License, v.
+ * 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server.subscriptions.service;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.transaction.Transactional;
+
+import org.springframework.stereotype.Service;
+
+import nu.yona.server.exceptions.YonaException;
+import nu.yona.server.subscriptions.entities.User;
+import nu.yona.server.subscriptions.service.migration.EncryptBuddyLastStatusChangeTime;
+
+@Service
+public class MigratePrivateUserDataService
+{
+	public static abstract class MigrationStep
```

Comment:
I always choose an abstract class as the default option, because it is easier to refactor later.